File: recursive/hanoi.c

#include<stdio.h>
void hanoi(char A,char B,char C,int n){
    if(n==1){
        printf("%c-->%c\n",A,C);
    }
    else{
        hanoi(A,C,B,n-1);
        printf("%c-->%c\n",A,C);
        hanoi(B,A,C,n-1);
    }
}
int main()
{
    int n;
    scanf("%d",&n);
    hanoi('A','B','C',n);
}

File: recursive/hanoi.py
count=0
def hanoi(A,B,C,n):
    global count
    if n==1:
        print("{}:{}-->{}".format(1,A,C))
        count+=1
    else:
        hanoi(A,C,B,n-1)
        print("{}:{}-->{}".format(n,A,C))
        count+=1
        hanoi(B,A,C,n-1)
n=int(input())
hanoi('A','B','C',n)
print(count)